integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala

```diff
@@ -495,15 +495,36 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
     // check duplicate columns and only 1 col left
     val distinctCols = noInvertedIdxColsProps.toSet
     // extract the no inverted index columns
+    val dictionaryInclude = tableProperties.getOrElse(CarbonCommonConstants.DICTIONARY_INCLUDE, "")
+      .split(",")
+    val sortColumns = tableProperties.getOrElse(CarbonCommonConstants.SORT_COLUMNS, "").split(",")
     fields.foreach(field => {
-      if (distinctCols.exists(x => x.equalsIgnoreCase(field.column))) {
+      if (distinctCols.exists(x => x.equalsIgnoreCase(field.column)) &&
+          validateColumnsForInvertedIndex(field, dictionaryInclude ++ sortColumns)) {
         noInvertedIdxCols :+= field.column
       }
     }
     )
     noInvertedIdxCols
   }
 
+
+  private def validateColumnsForInvertedIndex(field: Field,
+      dictionaryIncludeOrSortColumn: Array[String]): Boolean = {
+    val invertedIndexColumns = Array("date", "timestamp", "struct", "array")
```

Comment:
Struct, array will not have invertedindex, while data and timestamp will have invertedindex